alchemy.lic

```diff
@@ -0,0 +1,164 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#-------------
+=end
+
+custom_require.call(%w[common common-crafting common-items common-travel common-arcana])
+
+class Alchemy
+  include DRC
+  include DRCC
+  include DRCI
+  include DRCT
+  include DRCA
+
+  def initialize
+    arg_definitions = [
+      [
+        { name: 'target', regex: /\w+/, description: 'Ingredient or recipe.' },
+        { name: 'forage', regex: /forage/i, optional: true },
+        { name: 'prepare', regex: /prepare/i, optional: true },
+        { name: 'craft', regex: /craft/i, optional: true },
+        { name: 'herb_quantity_needed', regex: /\d+/, optional: true, description: 'Used to calculate required herbs to forage and process.' },
+        { name: 'debug', regex: /debug/i, optional: true }
+      ]
+    ]
+
+    args = parse_args(arg_definitions)
+
+    echo "*** incorrect arguments ***" if args.craft and (args.forage or args.prepare)
+    exit if args.craft and (args.forage or args.prepare)
+
+    settings = get_settings
+    @hometown = settings.hometown
+    @alchemy_herb_quantity = settings.alchemy_herb_quantity
+    @alchemy_herb_storage = settings.alchemy_herb_storage
+    @herb_container = settings.herb_container
+    @alchemy_prep_quantity = settings.alchemy_prep_quantity
+    @alchemy_forage_type = settings.alchemy_forage_type
+    @forage_override_room = settings.forage_override_room
+    @forage_override_town = settings.forage_override_town
+    @debug = args.debug
+    @herb_quantity_needed = args.herb_quantity_needed.to_i
+
+    ingredient = get_data('ingredients').ingredients.find{ |ingredient| ingredient['name'] =~ /#{args.target}/i } unless args.craft
+    # Set town for foraging
+    @forage_location = @forage_override_town || @forage_override_room || ingredient[@hometown]
+    echo ("@forage_location = #{@forage_location}") if @debug
+    @press_location = @forage_override_town || ingredient[@hometown]
+    echo ("@press_location = #{@press_location}") if @debug
+    recipe = get_data('recipes').crafting_recipes.find { |recipe| recipe['name'] =~ /#{args.target}/i } if args.craft
+    forage_ingredient(ingredient) if args.forage
```

Comment:
❓ Something seems off here. This line can run if `args.craft` is present, yet `ingredient` will be `nil`. Is that intended?

Reply:
Yes, intended. You call the script twice; once to gather and process and once to craft, if using this script to craft.

Reply:
That seems to me like this should be two separate scripts then: one to gather & process and one to craft. What do you think?

Reply:
Could just make it part of `;craft` I suppose. But then one could not run it standalone and select whatever recipe they want to make outside of training.

Reply:
I like the all-in-one approach.

Reply:
Removed craft and day/night check (not used presently).